tests/unit/plugins/modules/test_support_assist.py

```diff
@@ -0,0 +1,267 @@
+
+"""Unit Tests for support assist settings module on PowerScale"""
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+import pytest
+from mock.mock import MagicMock
+# pylint: disable=unused-import
+from ansible_collections.dellemc.powerscale.tests.unit.plugins.module_utils.shared_library.initial_mock \
+    import utils
+from ansible_collections.dellemc.powerscale.plugins.modules.support_assist import SupportAssist
+from ansible_collections.dellemc.powerscale.plugins.modules.support_assist import SupportAssistHandler
+from ansible_collections.dellemc.powerscale.plugins.modules.support_assist import main
+from ansible_collections.dellemc.powerscale.tests.unit.plugins.module_utils.mock_support_assist_api \
+    import MockSupportAssistApi
+from ansible_collections.dellemc.powerscale.tests.unit.plugins.module_utils.mock_api_exception \
+    import MockApiException
+from ansible_collections.dellemc.powerscale.tests.unit.plugins.module_utils.shared_library.powerscale_unit_base \
+    import PowerScaleUnitBase
+from ansible_collections.dellemc.powerscale.tests.unit.plugins.module_utils.mock_sdk_response \
+    import MockSDKResponse
+
+
+class TestSupportAssist(PowerScaleUnitBase):
+    support_assist_args = MockSupportAssistApi.SUPPORT_ASSIST_COMMON_ARGS
+
+    @pytest.fixture
+    def module_object(self):
+        return SupportAssist
+
+    def test_get_support_assist_details(self, powerscale_module_mock):
+        self.set_module_params(powerscale_module_mock,
+                               self.support_assist_args, {})
+        SupportAssistHandler().handle(powerscale_module_mock,
+                                      powerscale_module_mock.module.params)
+        powerscale_module_mock.support_assist_api.get_supportassist_settings.assert_called()
+
+    def test_get_support_assist_details_exception(self, powerscale_module_mock):
+        self.set_module_params(powerscale_module_mock,
+                               self.support_assist_args, {})
+        powerscale_module_mock.support_assist_api.get_supportassist_settings = MagicMock(
+            side_effect=MockApiException)
+        self.capture_fail_json_call(
+            MockSupportAssistApi.get_support_assist_settings_exception_response('get_details_exception'),
+            powerscale_module_mock, SupportAssistHandler)
+
+    def test_modify_support_assist_network_pool_response(self, powerscale_module_mock):
+        self.set_module_params(
+            powerscale_module_mock, self.support_assist_args,
+            {
+                "connection": {
+                    "network_pools": [
+                        {
+                            "pool_name": "subnet0:pool1",
+                            "state": "absent"
+                        },
+                        {
+                            "pool_name": "subnet0:pool5",
+                            "state": "present"
```

Comment:
Use variable if it repeated more than thrice to eliminate the sonar issues.

Reply:
used just once.